src/coreclr/jit/importer_vectorization.cpp

```diff
@@ -179,15 +256,21 @@ GenTree* Compiler::impExpandHalfConstEqualsSIMD(GenTreeLclVar* data, WCHAR* cns,
     //   vpxor    xmm1, xmm1, xmmword ptr[reloc @RWD16]
     //
 
+    if (cmpMode == OrdinalIgnoreCase)
+    {
+        // Apply ASCII-only ToLowerCase mask (bitwise OR 0x20 for all a-Z chars)
+        assert((toLowerVec1 != nullptr) && (toLowerVec2 != nullptr));
+        vec1 = gtNewSimdBinOpNode(GT_OR, simdType, vec1, toLowerVec1, baseType, simdSize, false);
+        vec2 = gtNewSimdBinOpNode(GT_OR, simdType, vec2, toLowerVec2, baseType, simdSize, false);
+    }
+
     // ((v1 ^ cns1) | (v2 ^ cns2)) == zero
     GenTree* xor1 = gtNewSimdBinOpNode(GT_XOR, simdType, vec1, cnsVec1, baseType, simdSize, false);
     GenTree* xor2 = gtNewSimdBinOpNode(GT_XOR, simdType, vec2, cnsVec2, baseType, simdSize, false);
     GenTree* orr  = gtNewSimdBinOpNode(GT_OR, simdType, xor1, xor2, baseType, simdSize, false);
     return gtNewSimdHWIntrinsicNode(TYP_BOOL, useSingleVector ? xor1 : orr, zero, niEquals, baseType, simdSize);
```

Comment:
What happens when `useSingleVector` is `true`. `xor2` and `orr` trees are just abandoned?

Reply:
Yes, but I assume it's a rare case, basically, 6.25% probability 😄 I was not sure it was worth it to add more control flow here or DEBUG_DESTROY